kernel/comps/block/src/device_id.rs

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::collections::btree_set::BTreeSet;
+use core::sync::atomic::{AtomicU32, Ordering};
+
+use device_id::DeviceId;
+use ostd::sync::Mutex;
+
+use crate::Error;
+
+/// The upper limit for block device major IDs (not included).
+const MAJOR_MAX: u32 = 512;
+
+const LAST_DYNAMIC_MAJOR: u32 = 254;
+
+static MAJORS: Mutex<BTreeSet<u32>> = Mutex::new(BTreeSet::new());
+
+/// Registers a major ID.
+///
+/// If the requested ID is 0, a dynamic ID will be registered.
+pub fn register_major(major: u32) -> Result<u32, Error> {
+    if major >= MAJOR_MAX {
```

Comment:
Move the definition of `MAJOR_MAX` into `register_major` because this is the only place that the const is used.